scripts/build.py

```diff
@@ -648,7 +724,19 @@ def generate_build_directory():
     save_hashes_as_json(HASHES_JSON, hashes)
 
     # Minify all template files copy them into build/templates/head.
-    build_files(TEMPLATES_DEV_DIR_CORE, TEMPLATES_STAGING_DIR, hashes)
+    if not os.path.isdir(TEMPLATES_STAGING_DIR):
+        print 'Creating new %s folder' % TEMPLATES_STAGING_DIR
+        build_files(TEMPLATES_DEV_DIR_CORE, TEMPLATES_STAGING_DIR, hashes)
+    else:
+        # Always rebuild HTML files.
+        build_HTML_files(TEMPLATES_DEV_DIR_CORE, TEMPLATES_STAGING_DIR, hashes)
+        new_files_list = get_new_files_from_directory(
+            TEMPLATES_DEV_DIR_CORE, TEMPLATES_OUT_DIR)
+        if new_files_list:
+            print 'Changes detected, re-generating %s' % TEMPLATES_OUT_DIR
+            rebuild_new_files(
+                TEMPLATES_DEV_DIR_CORE, TEMPLATES_STAGING_DIR, new_files_list,
+                hashes)
```

Comment:
Perhaps also do: else, print 'No changes detected. Using previously-built files.'

In general the more info and context you can give the developer, the better.

Reply:
Added comment!